### PR TITLE
OpenFilterPanelTemplate and SecurityFilterPanel

### PR DESCRIPTION
**Pull request by shaobinspire**

### CODE REVIEW

File: Applications/Spire/Include/Spire/Ui/SecurityFilterPanel.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   * @param query_model The model used to query matches.
+   * @param parent The parent widget.
+   */
+  SecurityFilterPanel* make_security_filter_panel(
+    std::shared_ptr<ComboBox::QueryModel> query_model, QWidget& parent);
+
+  /**
+   * Returns a new SecurityFilterPanel.
+   * @param query_model The model used to query matches.
+   * @param matches The list of secuirty values to match against.
```
Comment (spireken) [inline Applications/Spire/Include/Spire/Ui/SecurityFilterPanel.hpp:25]: Typo in security.

File: Applications/Spire/Include/Spire/Ui/OpenFilterPanelTemplate.hpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+       * @param title The title of the panel.
+       * @param parent The parent widget.
+       */
+      OpenFilterPanelTemplate(InputBoxBuilder input_box_builder, QString title,
+        QWidget& parent);
+
+      /**
+       * Constructs an OpenFilterPanelTemplate.
+       * @param input_box_builder The InputBoxBuilder to use.
+       * @param matches The list of secuirty values to match against.
```
Comment (spireken) [inline Applications/Spire/Include/Spire/Ui/OpenFilterPanelTemplate.hpp:51]: Typo.

**Review by spireken — CHANGES_REQUESTED** [review]

**Review by spireken — APPROVED** [review]

File: Applications/Spire/Source/Ui/OpenFilterPanel.cpp [hunk 3]
```diff
@@ -294,9 +294,10 @@ bool OpenFilterPanel::event(QEvent* event) {
   if(event->type() == QEvent::ShowToParent) {
     m_filter_panel->show();
     window()->activateWindow();
-    m_input_box->setFocus();
   } else if(event->type() == QEvent::HideToParent) {
     m_filter_panel->hide();
+  } else if(event->type() == QEvent::FocusIn) {
```
Comment (kranar) [inline Applications/Spire/Source/Ui/OpenFilterPanel.cpp:299]: Why not use a focus proxy?

**Review by kranar — CHANGES_REQUESTED** [review]

File: Applications/Spire/Source/Ui/OpenFilterPanel.cpp (hunk 3 above)
Comment (shaobinspire) [reply Applications/Spire/Source/Ui/OpenFilterPanel.cpp:299]: After testing, the focus proxy could break the Tab order. When the input box is set to the focus proxy of the OpenFilterPanel, the two Radio buttons on the top can't get focus by pressing TAB key.

**Pull request merged**